streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -432,6 +460,10 @@ public void onChange(final KafkaStreams.State newState, final KafkaStreams.State
             this.oldState = oldState;
             this.newState = newState;
             this.mapStates.put(newState, prevCount + 1);
+            if (this.closeOnChange) {
```

Comment:
For this to simulate trying to achieve a deadlock I think we need to to do the change on another thread. As it stands this thread would already own any locks, so it should always work regardless?